init-dotnet.sh

```diff
@@ -13,3 +13,5 @@ if [ $? != 0 ]; then
     echo "Failed to install dotnet using Arcade"
     exit $?
 fi
+
+export DOTNET_INSTALL_DIR=${_InitializeDotNetCli}
```

Comment:
It seems weird that that we need to explicitly set `DOTNET_INSTALL_DIR` ourselves on Unix based on what Arcade finds, but it looks like we expect something else to set it on Windows?

Reply:
The unix side will set the environment variable correctly because it sources the file. Therefore the return value `${_InitializeDotNetCli}` is valid. I cannot think of a better way on windows than capturing source-build's global env variable `DOTNET_TOOL_DIR` that is set during the build on windows.

Reply:
I can follow up with the source-build team to see what their suggestion is.

Reply:
Could you add comments about needing to respect overridden locations and where we expect the variable to come from?

Reply:
Removed the windows bit. Just using the return value on unix now.